paasta_tools/mesos_tools.py

```diff
@@ -615,3 +615,36 @@ def get_mesos_network_for_net(net):
         'host': 'HOST',
     }
     return docker_mesos_net_mapping.get(net, net)
+
+
+def get_mesos_task_count_by_slave(mesos_state, pool='default'):
+    """Get counts of running tasks per mesos slave. Also include separate count of chronos tasks
+
+    :param mesos_state: mesos state dict
+    :param pool: pool of slaves to return (None means all)
+    :returns: dict of {<slave_id>:{'count': <count>, 'slave': <slave_object>, 'chronos_count':<chronos_count>}}
+    """
+    all_mesos_tasks = get_running_tasks_from_active_frameworks('')  # empty string matches all app ids
+    if pool:
```

Comment:
right, but what I mean is the default value doesn't make sense as it currently stands. If I wanted to get **all** the pools, then I'd have to do ``get_mesos_task_count_by_slave(mesos_state, pool=None)``, which is odd.


Two suggestions here:

- Set the default parameter to be None, so that you get **all** pools unless you specify otherwise
```
def get_mesos_task_count_by_slave(mesos_state, pool=None)
```


- rename the parameter to ``pool_filter``, and have it as a function that returned ``True`` when you matched a pool that you're interested in, and ``False`` otherwise. The caller could then do something like:
```
get_mesos_task_count_by_slave(mesos_state, lambda pool: pool == 'default')
get_mesos_task_count_by_slave(mesos_state, lambda pool: pool == 'lucy')
get_mesos_task_count_by_slave(mesos_state, lambda pool: True) #match any
```

I kind of like the second one for it's flexibility, but I'll leave it with you to decide.